akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/internal/R2dbcOffsetStore.scala

```diff
@@ -236,6 +239,8 @@ private[projection] class R2dbcOffsetStore(
   // To avoid delete requests when no new offsets have been stored since previous delete
   private val idle = new AtomicBoolean(false)
 
+  private var validationObservers: immutable.Seq[R2dbcOffsetValidationObserver] = Nil
```

Comment:
Doesn't it need to be volatile or an atomic (or a copy method of some kind)?

Reply:
it's set once at startup, and as far as I can see it's not important if we would miss a few notifications. The purpose is not to track each and every offset, but see overall progress or what is going on. I can add a comment.